src/jmh/java/column/store/query/FilterForStringOperations.java

```diff
@@ -0,0 +1,91 @@
+package column.store.query;
+
+import column.store.api.column.*;
+import column.store.api.query.Query;
+import column.store.api.query.StringFilter;
+import column.store.api.read.Reader;
+import column.store.api.read.StringColumnReader;
+import column.store.util.BenchmarkUtil;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import static column.store.api.query.Filter.whereString;
+
+public class FilterForStringOperations {
+
+    private static final StringColumn SPAN_KIND = Column.forString("span.kind-string");
+    private static final StringColumn ENDPOINT_NAME = Column.forString("endpoint.name-string");
+    private static final StringColumn SERVLET_CONTEXT_NAME = Column.forString("servlet.context.name-string");
+    private static final StringColumn THREAD_NAME = Column.forString("thread.name-string");
+
+    private static final StringColumn[] STRING_COLUMNS = new StringColumn[] {
+            SPAN_KIND, ENDPOINT_NAME, SERVLET_CONTEXT_NAME, THREAD_NAME
+    };
+
+
+    private static final StringFilter IS_INTERNAL_SPANS = whereString(SPAN_KIND).is("internal");
+    private static final StringFilter CONTAINS_LOCALHOST_ENDPOINT = whereString(ENDPOINT_NAME).contains("localhost");
+    private static final StringFilter SERVLET_NAME_STARTS_WITH_EASYTRAVEL = whereString(ENDPOINT_NAME).startsWith("easyTravel");
+    private static final StringFilter THREAD_NAME_ENDS_WITH = whereString(ENDPOINT_NAME).endsWith("1172");
+
+
+    private static final StringFilter[] STRING_FILTERS = new StringFilter[] {
+            IS_INTERNAL_SPANS, CONTAINS_LOCALHOST_ENDPOINT, SERVLET_NAME_STARTS_WITH_EASYTRAVEL, THREAD_NAME_ENDS_WITH
+    };
+
+    @State(Scope.Thread)
+    public static class BenchState {
+        public Reader reader;
+        public Query query;
+        public StringColumn[] columns;
+        public Path source;
+
+        @Param({"parquet", "csv"})
+        private String readerType;
+        @Param({"spans", "logs"})
```

Comment:
IMO we should have different bechmarks for spans and logs, as the data is quite different, contains different columns, etc.

Reply:
ok